components/ai_chat/resources/page/state/active_chat_provider.tsx

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) 2024 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import * as React from 'react'
+import getAPI, * as API from '../api'
+import { useRoute } from '$web-common/useRoute'
+import { useAIChat } from './ai_chat_context'
+
+export const tabAssociatedChatId = 'tab'
+
+export interface SelectedChatDetails {
+  selectedConversationId: string | undefined
+  updateSelectedConversationId: (conversationId: string | undefined) => void
+  conversationHandler: API.ConversationHandlerRemote
+  callbackRouter: API.ConversationUICallbackRouter
+  newConversation: () => void
```

Comment:
semi-nit: this doesn't sound enough like an action. When I see it in the UI code, it can be confused for a newConversation variable. Suggestions: onNewConversation, handleNewConversation, createNewConversation